packages/tpetra/core/src/Tpetra_MultiVector_def.hpp

```diff
@@ -3996,18 +3997,39 @@ namespace Tpetra {
       const LO C_lclNumRows = C_tmp->getLocalLength ();
       const LO C_numVecs = C_tmp->getNumVectors ();
       auto C_lcl = C_tmp->getLocalViewDevice ();
-      auto C_sub = Kokkos::subview (C_lcl,
-                                    std::make_pair (LO (0), C_lclNumRows),
-                                    std::make_pair (LO (0), C_numVecs));
-      const char ctransA = (transA == Teuchos::NO_TRANS ? 'N' :
-                            (transA == Teuchos::TRANS ? 'T' : 'C'));
-      const char ctransB = (transB == Teuchos::NO_TRANS ? 'N' :
-                            (transB == Teuchos::TRANS ? 'T' : 'C'));
+
+      const char ctransA = (transA == NO_TRANS ? 'N' :
+                            (transA == TRANS ? 'T' : 'C'));
+      const char ctransB = (transB == NO_TRANS ? 'N' :
+                            (transB == TRANS ? 'T' : 'C'));
       const impl_scalar_type alpha_IST (alpha);
 
-      ProfilingRegion regionGemm ("Tpetra::MV::multiply-call-gemm");
-      KokkosBlas::gemm (&ctransA, &ctransB, alpha_IST, A_sub, B_sub,
-                        beta_local, C_sub);
+      const size_t zero = 0;
```

Comment:
This is harmless, but you don't need it.  `Kokkos::subview(X, ALL (), 0)` works perfectly fine.